src/parser/PathQuery.h

```diff
@@ -0,0 +1,83 @@
+// Copyright 2024, University of Freiburg
+// Chair of Algorithms and Data Structures
+// Authors: Johannes Herrmann <johannes.r.herrmann(at)gmail.com>
+//          Christoph Ullinger <ullingec@informatik.uni-freiburg.de>
+
+#pragma once
+
+#include "parser/MagicServiceQuery.h"
+
+class SparqlTriple;
+
+namespace parsedQuery {
+
+class PathSearchException : public std::exception {
+  std::string message_;
+
+ public:
+  explicit PathSearchException(const std::string& message)
+      : message_(message) {}
+  const char* what() const noexcept override { return message_.data(); }
+};
+
+// The PathQuery object holds intermediate information for the PathSearch.
+// The PathSearchConfiguration requires concrete Ids. The vocabulary from the
+// QueryPlanner is needed to translate the TripleComponents to ValueIds.
+// Also, the members of the PathQuery have defaults and can be set after
+// the object creation, simplifying the parsing process. If a required
+// value has not been set during parsing, the method 'toPathSearchConfiguration'
+// will throw an exception.
+// All the error handling for the PathSearch happens in the PathQuery object.
+// Thus, if a PathSearchConfiguration can be constructed, it is valid.
+struct PathQuery : MagicServiceQuery {
+  std::vector<TripleComponent> sources_;
+  std::vector<TripleComponent> targets_;
+  std::optional<Variable> start_;
+  std::optional<Variable> end_;
+  std::optional<Variable> pathColumn_;
+  std::optional<Variable> edgeColumn_;
+  std::vector<Variable> edgeProperties_;
+  PathSearchAlgorithm algorithm_;
+
+  bool cartesian_ = true;
+  std::optional<uint64_t> numPathsPerTarget_ = std::nullopt;
+
+  PathQuery() = default;
+  PathQuery(PathQuery&& other) noexcept = default;
+  PathQuery(const PathQuery& other) noexcept = default;
+  PathQuery& operator=(const PathQuery& other) = default;
+  PathQuery& operator=(PathQuery&& a) noexcept = default;
+  ~PathQuery() noexcept override = default;
```

Comment:
Same comment here as below, can't we delete these and they are explicitly defaulted?